core/src/main/scala/org/apache/spark/launcher/LauncherBackend.scala

```diff
@@ -110,12 +146,14 @@ private[spark] abstract class LauncherBackend {
     override def close(): Unit = {
       try {
         super.close()
+        if (stopFlag) {
```

Comment:
This seems to be overloading the meaning of `stopFlag`. Does it mean "stop the app when the launcher goes down", or "stop the app when the launcher is closed"? This code is implementing the latter, while the earlier shutdown hook is implementing the former. I don't think the latter is desired in the API; there is already an explicit API for that.